tests/test_steps.py

```diff
@@ -0,0 +1,123 @@
+from pytest import fail
+
+from storey import build_flow, Source, Map
+from storey.dtypes import FlowError, Event
+from storey.steps import Flatten, Sample, Assert, ForEach
+from time import sleep
+
+
+def test_assert():
+    try:
+        controller = build_flow(
+            [Source(), Assert(lambda event: event > 10, raise_on_exception=True)]
+        ).run()
+        controller.emit(1)
+        controller.await_termination()
+    except FlowError:
+        return True
+
+    fail("User defined assert not failing", False)
+
+
+def test_sample_by_count():
+    class CountUntil:
+        def __init__(self, max: int):
+            self.max = max
+            self.count = 0
+
+        def __call__(self, event):
+            self.count += 1
+            return self.count <= self.max
+
+    controller = build_flow(
+        [Source(), Assert(CountUntil(5)), Sample(5), Assert(CountUntil(1))]
```

Comment:
Using an Assert step is not good enough. If, for example, Sample had a bug where it emitted nothing, this test would still pass. Better use `Reduce` and `assert` as in other tests.